source/d1formats/d1cl2.cpp

```diff
@@ -443,7 +445,7 @@ bool D1Cl2::writeFileData(D1Gfx &gfx, QFile &outFile, bool isClx, const QString
     }
 
     // calculate sub header size
-    int subHeaderSize = SUB_HEADER_SIZE;
+    int subHeaderSize = isClx ? SUB_HEADER_SIZE : 6;
```

Comment:
This appears to be backwards.
```suggestion
    int subHeaderSize = !isClx ? SUB_HEADER_SIZE : 6;
```

Reply:
Thanks!